src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1342,6 +1278,27 @@ static bool processContextAttributeVector (ContextElement*
 
     }
 
+    /* Special processing in the case of DELETE, to avoid "deleting too much" in the case of using
+     * metadata ID. If metadata ID feature were removed this ugly piece of code wouldn't be needed ;)
+     * FIXME P3: note that in the case of Active/Active configuration it could happen that
+     * the attrsName get desyncrhonized, e.g. starting situation is A-ID1 and A-ID2, node
+     * #1 process DELETE A-ID1 and node #2 process DELETE A-ID2, each one thinking than the
+     * other A-IDx copy will be there at the end, thus nobody includes it in the toPull array
+     */
+    if (strcasecmp(action.c_str(), "delete") == 0)
+    {
+      for (std::map<string, unsigned int>::iterator it = deletedAttributesCounter.begin(); it != deletedAttributesCounter.end(); ++it)
+      {
+          std::string  attrName     = it->first;
+          unsigned int deletedTimes = it->second;
+
+          if (howManyAttrs(attrs, attrName) <= deletedTimes)
+          {
+            toPull->append(attrName);
+          }
+      }
+    }
+
 #if 0
```

Comment:
This #if 0 is a pending refactor. I would like to have a overall refactor of ugly things in mongoBackend, but I prefer to defer it to a "calm" moment (currently, there is a lot of activity here, some pending PR before closing 0.21.0 yet...).